tests/phpunit/CRM/Civicase/Service/CaseCategoryInstanceTest.php

```diff
@@ -0,0 +1,143 @@
+    // We generate two different category instance types.
+    $categoryInstanceTypeOne = $this->createRandomCategoryInstanceType();
+    $categoryInstanceTypeTwo = $this->createRandomCategoryInstanceType();
+    // And two random category instances associated with them.
+    $categoryInstanceOne = CaseCategoryInstanceFabricator::fabricate([
+      'instance_id' => $categoryInstanceTypeOne['value'],
+    ]);
+    $categoryInstanceTwo = CaseCategoryInstanceFabricator::fabricate([
+      'instance_id' => $categoryInstanceTypeTwo['value'],
+    ]);
+
+    // We call the method with categoryInstanceTypeOne name as a parameter.
+    $categoryInstances = (new CaseCategoryInstanceService())->getCaseCategoryInstances(
+      $categoryInstanceTypeOne['name']
+    );
+
+    // We verify that only category one is on the results.
+    $this->assertCount(1, $categoryInstances);
+    $this->assertArrayHasKey($categoryInstanceOne['id'], $categoryInstances);
+    $this->assertArrayNotHasKey($categoryInstanceTwo['id'], $categoryInstances);
+    $this->assertEquals(
+      $categoryInstanceOne['category_id'],
+      $categoryInstances[$categoryInstanceOne['id']]->category_id
+    );
+    $this->assertEquals(
+      $categoryInstanceOne['instance_id'],
+      $categoryInstances[$categoryInstanceOne['id']]->instance_id
+    );
+  }
+
+  /**
+   * Test the correct creation of an instance for an existing category.
+   */
+  public function testAssignInstanceForExistingCaseCategoriesWithoutAnInstance() {
+    // Generate a random category.
+    $categoryWithoutInstance = CaseCategoryFabricator::fabricate();
+    // And a category instance, not related with the previous.
+    CaseCategoryInstanceFabricator::fabricate();
+    $categoryInstanceService = new CaseCategoryInstanceService();
+
+    // We assert that this category has no instance.
+    $categoryInstances = $categoryInstanceService->getCaseCategoryInstances();
+    $this->assertFalse($this->categoryHasInstance($categoryWithoutInstance, $categoryInstances));
+
+    // We call the method for creating the corresponding instance.
+    $categoryInstanceService->assignInstanceForExistingCaseCategories();
+
+    // And we check that was correctly created.
+    $categoryInstances = $categoryInstanceService->getCaseCategoryInstances();
+    $this->assertTrue($this->categoryHasInstance($categoryWithoutInstance, $categoryInstances));
+  }
+
+  /**
+   * Check if the given category has an instance on the array received.
+   */
+  private function categoryHasInstance($caseCategory, $caseCategoryInstances) {
+    foreach ($caseCategoryInstances as $caseCategoryInstance) {
+      if ($caseCategoryInstance->category_id == $caseCategory['value']) {
+        return TRUE;
+      }
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * Generates a random category instance type.
+   *
+   * @return array
+   *   Category instance type details.
+   */
+  private function createRandomCategoryInstanceType() {
```

Comment:
Can we rename to `createCategoryInstanceType` or  `generateCategoryInstanceType` ?. That is what it is doing in actual reality. It's not really doing anything random.

Reply:
Ok, sounds good. I used `random` in the sense of `fake`, but it is adding no real information to the functionality of the method. Done here: https://github.com/compucorp/uk.co.compucorp.civicase/pull/657/commits/1fd687a939ffede8d98d416c1668e021da33e1a1